app/workers/subscription_content_worker.rb

```diff
@@ -1,32 +1,48 @@
 class SubscriptionContentWorker
   include Sidekiq::Worker
 
-  def perform(content_change_id)
+  def perform(content_change_id, batch_size = 1000)
     content_change = ContentChange.find(content_change_id)
-    queue_delivery_to_subscribers(content_change)
+    queue_delivery_to_subscribers(content_change, batch_size: batch_size)
     queue_delivery_to_courtesy_subscribers(content_change)
     content_change.mark_processed!
   end
 
 private
 
-  def queue_delivery_to_subscribers(content_change)
-    subscriptions_for(content_change: content_change).find_in_batches do |group|
-      columns = %i(content_change_id subscription_id)
+  def queue_delivery_to_subscribers(content_change, batch_size: 1000)
+    content_change_id = content_change.id
+    batch = []
 
-      content_change_id = content_change.id
-      records = group.pluck(:id).map do |subscription_id|
+    grouped_subscription_ids_per_subscriber(content_change).each_value do |subscription_ids|
+      records = subscription_ids.map do |subscription_id|
         [content_change_id, subscription_id]
       end
 
-      begin
-        SubscriptionContent.import!(columns, records)
-      rescue StandardError => ex
-        Raven.capture_exception(ex, tags: { version: 2 })
-      end
+      batch.concat(records)
 
-      ImmediateEmailGenerationWorker.perform_async
+      if batch.size >= batch_size
+        import_subscription_contents_batch(batch)
+        batch.clear
+      end
     end
+
+    import_subscription_contents_batch(batch) unless batch.empty?
+  end
+
+  def import_subscription_contents_batch(batch)
+    columns = %i(content_change_id subscription_id)
+    SubscriptionContent.import!(columns, batch)
+    ImmediateEmailGenerationWorker.perform_async
+  end
+
+  def grouped_subscription_ids_per_subscriber(content_change)
+    subscriptions_for(content_change: content_change)
+      .pluck(:subscriber_id, :id)
+      .group_by(&:first)
+      .each_with_object({}) do |(subscriber_id, rows), hash|
+        hash[subscriber_id] = rows.map(&:last)
+      end
```

Comment:
You could possibly do more of this in postgres with a group(:subscriber_id) and pluck the array agg of the ids

Reply:
Good idea!